From 0159e28be2387103ec3380d81af1b73667130c5e Mon Sep 17 00:00:00 2001
From: Cody Peter Mello <cody.mello@joyent.com>
Date: Thu, 18 May 2017 21:27:13 +0000
Subject: [PATCH] joyent/node-smartdc-auth#12 KeyRingHomeDirPlugin.listKeys()
 invokes callback twice on failure Reviewed by: Trent Mick
 <trent.mick@joyent.com> Approved by: Trent Mick <trent.mick@joyent.com>

---
 lib/kr-homedir.js | 4 +++-
 package.json      | 2 +-
 2 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/lib/kr-homedir.js b/lib/kr-homedir.js
index 2d24fd4..9843426 100644
--- a/lib/kr-homedir.js
+++ b/lib/kr-homedir.js
@@ -1,5 +1,5 @@
 /*
- * Copyright 2016 Joyent, Inc., All rights reserved.
+ * Copyright 2017 Joyent, Inc., All rights reserved.
  *
  * Permission is hereby granted, free of charge, to any person obtaining a copy
  * of this software and associated documentation files (the "Software"), to deal
@@ -78,7 +78,9 @@ KeyRingHomeDirPlugin.prototype.listKeys = function (cb) {
 	}, function (err) {
 		if (err && keys.length < 1) {
 			cb(err);
+			return;
 		}
+
 		cb(null, keys);
 	});
 };
diff --git a/package.json b/package.json
index b58f8b4..cb2e44f 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
     "name": "smartdc-auth",
     "description": "Joyent Authentication Library",
-    "version": "2.5.3",
+    "version": "2.5.4",
     "author": "Joyent (joyent.com)",
     "contributors": [
         "Mark Cavage",
-- 
2.21.0

